src/modify.js

```diff
@@ -6,25 +6,69 @@ const config = require("./config").config;
 
 const isImportMap = config && config.manifestFormat === "importmap";
 
-function getMapFromManifest(manifest) {
-  return isImportMap ? manifest.imports : manifest.sofe.manifest;
+function getScopeFromManifest(manifest, scope = "imports") {
+  if (!isImportMap) {
+    if (scope !== "imports") {
+      throw new Error("Sofe implementations can only support imports");
+    }
+    return manifest.sofe.manifest;
+  } else {
+    return manifest[scope] || {};
+  }
 }
 
 function getEmptyManifest() {
-  return isImportMap ? { imports: {} } : { sofe: { manifest: {} } };
+  return isImportMap ? { imports: {}, scopes: {} } : { sofe: { manifest: {} } };
 }
+const deepCopy = (json) => JSON.parse(JSON.stringify(json));
 
-exports.getEmptyManifest = getEmptyManifest;
+exports.modifyMultipleServices = function (env, newImports) {
+  return modifyMultiple(env, newImports, "imports");
+};
 
 exports.modifyService = function (env, serviceName, url, remove) {
+  return modifyKeyValue(env, serviceName, url, remove, "imports");
+};
+
+exports.modifyMultipleScopes = function (env, newImports) {
+  return modifyMultiple(env, newImports, "scopes");
+};
+
+exports.modifyScope = function (env, serviceName, url, remove) {
+  return modifyKeyValue(env, serviceName, url, remove, "scopes");
+};
+
+function modifyMultiple(env, values, manifestScope) {
+  return modifyLock(env, (json) => {
+    const imports = getScopeFromManifest(json, manifestScope);
+    imports[manifestScope] = values;
```

Comment:
This appears to be wrong in the context of import map scopes. The `imports` key and `scopes` keys are quite different from each other and I think it makes sense to keep their implementations separate.

```json
{
  "imports": {
    "foo": "https://cdn.com/foo.js"
  },
  "scopes": {
    "https://cdn.com/": {
      "bar": "https://cdn.com/bar.js"
    }
  }
}
```

Notice how the values in the `imports` key are strings, but the the values in `scopes` key are objects.